### PR TITLE
ci: add e2e tests for TrafficRoute

### PR DESCRIPTION
**Pull request by yskopets**
### Summary

* add `e2e` tests for `TrafficRoute`
  * both `universal` and `k8s` modes
  * both with mTLS and without mTLS

### CODE REVIEW

File: Makefile.e2e.mk [hunk 1]
```diff
@@ -170,32 +326,145 @@ apply/example/minikube/mtls: ## Minikube: enable mTLS
...
+	@echo
+	@echo "Deploying example setup for TrafficRoute ..."
+	@echo
+	kubectl apply -f tools/e2e/examples/minikube/kuma-routing/
+	kubectl wait --timeout=60s --for=condition=Available -n kuma-example deployment/kuma-example-web
+	kubectl wait --timeout=60s --for=condition=Ready -n kuma-example pods -l app=kuma-example-web
+	kubectl wait --timeout=60s --for=condition=Available -n kuma-example deployment/kuma-example-backend-v1
+	kubectl wait --timeout=60s --for=condition=Ready -n kuma-example pods -l app=kuma-example-backend,version=v1
+	kubectl wait --timeout=60s --for=condition=Available -n kuma-example deployment/kuma-example-backend-v2
+	kubectl wait --timeout=60s --for=condition=Ready -n kuma-example pods -l app=kuma-example-backend,version=v2
```
Comment (hutchic) [inline Makefile.e2e.mk]: (don't change this it's fine) I'm a fan of
```
while [[ "$(kubectl get pod --all-namespaces | grep -v Running | grep -v Completed | wc -l)" != 1 ]]; do
  kubectl get pod --all-namespaces -o wide
  echo "waiting for K8s to be ready"
  sleep 10;
done
```

It's less specific but sometimes that's what's needed ;)

File: tools/e2e/examples/docker-compose/docker-compose.yaml [hunk 2]
```diff
@@ -214,9 +235,222 @@ services:
...
+  #                          \
+  #                            -> backend-v2  :  version: v2, env=intg
+
+  #
+  # Auxiliary service (container) for sharing Linux network namespace
+  # between Example Web and Kuma Sidecar.
+  #
+  kuma-example-web-namespace:
+    image: busybox:1.31.0
+    command: ["sleep", "infinity"]
```
Comment (hutchic) [inline tools/e2e/examples/docker-compose/docker-compose.yaml]: TIL 👍

File: tools/e2e/examples/docker-compose/docker-compose.yaml [hunk 3]
```diff
@@ -214,9 +235,222 @@ services:
...
+    - kuma-control-plane
+    restart: on-failure
+
+  #
+  # Example Backend v1 deployed into Kuma mesh.
+  #
+  # Notice that we're using Envoy as a simple HTTP server with predefined responses.
+  #
+  kuma-example-backend-v1:
+    image: envoyproxy/envoy-alpine:v1.12.1
```
Comment (hutchic) [inline tools/e2e/examples/docker-compose/docker-compose.yaml]:
```suggestion
    image: envoyproxy/envoy-alpine:${ENVOY_VERSION:-v1.12.1}
```

File: Makefile.e2e.mk (hunk 1 above)
Comment (yskopets) [reply Makefile.e2e.mk]: This `while` loop doesn't account for a situation where `Pods` are not created at all

File: tools/e2e/examples/docker-compose/docker-compose.yaml (hunk 3 above)
Comment (yskopets) [reply tools/e2e/examples/docker-compose/docker-compose.yaml]: `${ENVOY_VERSION:-v1.12.1}` would be useful if we were testing different version of `envoy`, but it's not the case here (`envoy` is used as a sample HTTP server).

Trying to minimize amount of YAML developers will have to read, understand and maintain

**Pull request merged**